### PR TITLE
drop support for old-style Merkle txn root commitment

### PR DESCRIPTION
**Pull request by zeldovich**
This PR removes support for the Merkle transaction commitments that we used a long time ago (well before mainnet launch).  This will also help add support for a new form of Merkle tree commitment for transactions (PR #1879).

### CODE REVIEW

**Review by tsachiherman — CHANGES_REQUESTED** [review]
One small file that you've missed:
`./test/testdata/consensus/catchpointtestingprotocol.json`

**Comment by zeldovich** [comment]
> One small file that you've missed:
> `./test/testdata/consensus/catchpointtestingprotocol.json`

Thanks for catching that!  I was indeed hoping to smoke out potential issues by seeing what Travis finds.

File: ledger/ledger_test.go [hunk 1]
```diff
@@ -364,7 +371,10 @@ func TestLedgerBlockHeaders(t *testing.T) {
 func TestLedgerSingleTx(t *testing.T) {
 	a := require.New(t)
 
-	genesisInitState, initSecrets := testGenerateInitState(t, protocol.ConsensusV7)
+	// V15 is the earliest protocol version in active use.
+	// The genesis for betanet and testnet is at V15
+	// The genesis for mainnet is at V17
+	genesisInitState, initSecrets := testGenerateInitState(t, protocol.ConsensusV15)
```
Comment (tsachiherman) [inline ledger/ledger_test.go:377]: that's good observation; V7 is where Adam stopped with the consensus parameter folding.

**Comment by tsachiherman** [comment]
@nicholasguoalgorand - I'm adding you so that you'll be aware of the changes in here. You don't have to approve this one, just go over and make sure you're familiar with the changes.

**Comment by tsachiherman** [comment]
it looks like travis fails due to "make lint". Could you try to run this ?

**Comment by zeldovich** [comment]
@tsachiherman , looks like I found all of the references to old protocols that happened to use the old Merkle-tree commitment for transactions.  (I did not realize our tests were still exercising that code!)  Feel free to merge whenever you're happy with it.

**Review by tsachiherman — APPROVED** [review]

**Pull request merged**